src/domain/community/user/user.resolver.fields.ts

```diff
@@ -27,6 +28,15 @@ export class UserResolverFields {
     return await this.userService.getAgent(user.id);
   }
 
+  @ResolveField('preferences', () => [IUserPreference], {
+    nullable: false,
+    description: 'The preferences for this user',
+  })
+  @Profiling.api
+  preferences(@Parent() user: User): Promise<IUserPreference[]> {
+    return this.userService.getPreferences(user.id);
```

Comment:
am pretty sure this method is async and should be awaited, which also means the field resolver should be async